src/game/inventry.c

```diff
@@ -681,98 +681,58 @@ int32_t Display_Inventory(int inv_mode)
                 // page 1: select level
-                // TODO Placeholder for select level. Do new game.
-                Savegame_InitStartCurrentInfo();
-                return GF_START_GAME | g_InvExtraData[IED_LEVEL_NUM];
-            } else if (g_InvExtraData[IED_PAGE_NUM] == PASSPORT_PAGE_2) {
+                return GF_SELECT_GAME | g_GameInfo.select_level_num;
+            } else if (g_GameInfo.passport_page == PASSPORT_PAGE_2) {
                 // page 2: restart level
                 return GF_RESTART_GAME | g_CurrentLevel;
             } else {
                 // page 3: exit game
                 return GF_EXIT_TO_TITLE;
             }
         } else {
-            if (g_InvExtraData[IED_PAGE_NUM] == PASSPORT_PAGE_1
-                && g_InvExtraData[IED_PASSPORT_MODE]
-                    == PASSPORT_MODE_SHOW_SAVES) {
+            if (g_GameInfo.passport_page == PASSPORT_PAGE_1
+                && g_GameInfo.passport_mode == PASSPORT_MODE_SHOW_SAVES) {
                 // page 1: load game
-                return GF_START_SAVED_GAME | g_InvExtraData[IED_SAVEGAME_NUM];
+                return GF_START_SAVED_GAME | g_GameInfo.current_save_slot;
             } else if (
-                g_InvExtraData[IED_PAGE_NUM] == PASSPORT_PAGE_1
-                && g_InvExtraData[IED_PASSPORT_MODE]
-                    == PASSPORT_MODE_SELECT_LEVEL) {
+                g_GameInfo.passport_page == PASSPORT_PAGE_1
+                && g_GameInfo.passport_mode == PASSPORT_MODE_SELECT_LEVEL) {
                 // page 1: select level
-                // TODO Placeholder for select level. Do new game.
-                Savegame_InitStartCurrentInfo();
-                return GF_START_GAME | g_InvExtraData[IED_LEVEL_NUM];
-            } else if (g_InvExtraData[IED_PAGE_NUM] == PASSPORT_PAGE_2) {
-                // page 1: save game, or new game in gym
+                return GF_SELECT_GAME | g_GameInfo.select_level_num;
+            } else if (g_GameInfo.passport_page == PASSPORT_PAGE_2) {
+                // page 2: new game in gym
                 if (g_CurrentLevel == g_GameFlow.gym_level_num) {
-                    switch (g_InvExtraData[IED_SAVEGAME_NUM]) {
-                    case 0:
-                        g_GameInfo.bonus_flag = 0;
-                        break;
-                    case 1:
-                        g_GameInfo.bonus_flag = GBF_NGPLUS;
-                        break;
-                    case 2:
-                        g_GameInfo.bonus_flag = GBF_JAPANESE;
-                        break;
-                    case 3:
-                        g_GameInfo.bonus_flag = GBF_JAPANESE | GBF_NGPLUS;
-                        break;
-                    }
-                    Savegame_InitStartCurrentInfo();
+                    Savegame_InitCurrentInfo();
                     return GF_START_GAME | g_GameFlow.first_level_num;
+                    // page 2: save game
```

Comment:
Weird placement of the comment, was it meant to be inside `} else {`?

Reply:
Whoops should be down one more line.